Source/WarpX.cpp

```diff
@@ -323,6 +324,17 @@ WarpX::ReadParameters ()
         pp.query("restart", restart_chkfile);
     }
 
+    {
+        ParmParse pp("interpolation");
+        pp.query("nox", nox);
+        pp.query("noy", noy);
+        pp.query("noz", noz);
+        pp.query("match_shape_factors", match_shape_factors);
+        AMREX_ALWAYS_ASSERT_WITH_MESSAGE( nox == noy and nox == noz ,
```

Comment:
`and` and `or` are not supported by default by all C++ compilers (notably MSVC):
```suggestion
        AMREX_ALWAYS_ASSERT_WITH_MESSAGE( nox == noy && nox == noz ,
```

Reply:
Careful: no need to enforce `noy` in a 2D3V simulation :)

Reply:
I don't think I wrote this line, actually - it looks like it's already there in older versions of WarpX. I agree that it should be changed, but should that be a separate PR?

Reply:
Oh right, moved from below.